src/main/java/net/minecraftforge/fml/relauncher/FMLSecurityManager.java

```diff
@@ -20,7 +20,7 @@ public void checkPermission(Permission perm)
             String callingClass = classContexts.length > 3 ? classContexts[4].getName() : "none";
             String callingParent = classContexts.length > 4 ? classContexts[5].getName() : "none";
             // FML is allowed to call system exit and the Minecraft applet (from the quit button)
-            if (!(callingClass.startsWith("net.minecraftforge.fml.") || ( "net.minecraft.client.Minecraft".equals(callingClass) && "net.minecraft.client.Minecraft".equals(callingParent)) || ("net.minecraft.server.dedicated.DedicatedServer".equals(callingClass) && "net.minecraft.server.MinecraftServer".equals(callingParent))))
+            if (!(callingClass.startsWith("net.minecraftforge.fml.") || "net.minecraft.server.dedicated.ServerHangWatchdog$1".equals(callingClass) || "net.minecraft.server.dedicated.ServerHangWatchdog".equals(callingClass) || ( "net.minecraft.client.Minecraft".equals(callingClass) && "net.minecraft.client.Minecraft".equals(callingParent)) || ("net.minecraft.server.dedicated.DedicatedServer".equals(callingClass) && "net.minecraft.server.MinecraftServer".equals(callingParent))))
             {
```

Comment:
@simon816 but there expression like:("net.minecraft.server.dedicated.DedicatedServer".equals(callingClass) && "net.minecraft.server.MinecraftServer".equals(callingParent)) thi can't be stored in a hashmap because the AND operator